src/Generating/StructGen.cpp

```diff
@@ -198,65 +198,69 @@ int cStructGenTrees::GetNumTrees(
-			case biJungleM:              Add = 120; break;
-			case biJungleEdgeM:          Add =  90; break;
-			case biBirchForestM:         Add =  30; break;
-			case biBirchForestHillsM:    Add =  20; break;
-			case biRoofedForestM:        Add =  40; break;
-			case biColdTaigaM:           Add =  30; break;
-			case biMegaSpruceTaiga:      Add =  30; break;
-			case biMegaSpruceTaigaHills: Add =  30; break;
-			case biExtremeHillsPlusM:    Add =   4; break;
-			case biSavannaM:             Add =   8; break;
-			case biSavannaPlateauM:      Add =  12; break;
-			case biMesaBryce:            Add =   4; break;
-			case biMesaPlateauFM:        Add =  12; break;
-			case biMesaPlateauM:         Add =  12; break;
+			case biOcean:                return 2;
+			case biDesert:               return 0;
+			case biPlains:               return 1;
+			case biExtremeHills:         return 3;
+			case biForest:               return 30;
+			case biTaiga:                return 30;
+			case biSwampland:            return 8;
+			case biIcePlains:            return 1;
+			case biIceMountains:         return 1;
+			case biMushroomIsland:       return 3;
+			case biMushroomShore:        return 3;
+			case biForestHills:          return 20;
+			case biTaigaHills:           return 20;
+			case biExtremeHillsEdge:     return 5;
+			case biJungle:               return 120;
+			case biJungleHills:          return 90;
+			case biJungleEdge:           return 90;
+			case biBirchForest:          return 30;
+			case biBirchForestHills:     return 20;
+			case biRoofedForest:         return 50;
+			case biColdTaiga:            return 20;
+			case biColdTaigaHills:       return 15;
+			case biMegaTaiga:            return 30;
+			case biMegaTaigaHills:       return 25;
+			case biExtremeHillsPlus:     return 3;
+			case biSavanna:              return 8;
+			case biSavannaPlateau:       return 12;
+			case biMesa:                 return 2;
+			case biMesaPlateauF:         return 8;
+			case biMesaPlateau:          return 8;
+			case biSunflowerPlains:      return 1;
+			case biDesertM:              return 0;
+			case biExtremeHillsM:        return 4;
+			case biFlowerForest:         return 30;
+			case biTaigaM:               return 30;
+			case biSwamplandM:           return 8;
+			case biIcePlainsSpikes:      return 1;
+			case biJungleM:              return 120;
+			case biJungleEdgeM:          return 90;
+			case biBirchForestM:         return 30;
+			case biBirchForestHillsM:    return 20;
+			case biRoofedForestM:        return 40;
+			case biColdTaigaM:           return 30;
+			case biMegaSpruceTaiga:      return 30;
+			case biMegaSpruceTaigaHills: return 30;
+			case biExtremeHillsPlusM:    return 4;
+			case biSavannaM:             return 8;
+			case biSavannaPlateauM:      return 12;
+			case biMesaBryce:            return 4;
+			case biMesaPlateauFM:        return 12;
+			case biMesaPlateauM:         return 12;
+			default:                     return 0;
```

Comment:
Is it right to have a default here? Ideally we should just list them all, and then the compiler will warn us when a new biome is added, so we can specify the tree level.

At least the default should assert or something.

Reply:
Well there are 80 biomes and only ~50 cases here so the defaulting behaviour must be intended.

Reply:
I think probably we should handle the rest of these, unless any new biomes are expected not to have trees,